contracts/DandelionOrg.sol

```diff
@@ -0,0 +1,382 @@
+    string constant private ERROR_MISSING_CACHE = "DANDELION_MISSING_CACHE";
+
+    bool constant private TOKEN_TRANSFERABLE = false;
+    uint8 constant private TOKEN_DECIMALS = uint8(18);
+    uint256 constant private TOKEN_MAX_PER_ACCOUNT = uint256(0);
+    uint64 constant private DEFAULT_FINANCE_PERIOD = uint64(30 days);
+
+    bytes32 constant private REDEMPTIONS_APP_ID = apmNamehash("redemptions");
+    bytes32 constant private TOKEN_REQUEST_APP_ID = apmNamehash("token-request");
+    bytes32 constant private TIME_LOCK_APP_ID = apmNamehash("time-lock");
+
+    struct Cache {
+        address dao;
+        address tokenManager;
+        address agentOrVault;
+        address finance;
+        address tokenRequest;
+        address redemptions;
+        address timeLock;
+        address dandelionVoting;
+        bool agentAsVault;
+    }
+
+    mapping (address => Cache) internal cache;
+
+    constructor(DAOFactory _daoFactory, ENS _ens, MiniMeTokenFactory _miniMeFactory, IFIFSResolvingRegistrar _aragonID)
+        BaseTemplate(_daoFactory, _ens, _miniMeFactory, _aragonID)
+        public
+    {
+        _ensureAragonIdIsValid(_aragonID);
+        _ensureMiniMeFactoryIsValid(_miniMeFactory);
+    }
+
+    /**
+    * @dev Create a new MiniMe token and deploy a Dandelion Org DAO. This function does not allow Payroll
+    *      to be setup due to gas limits.
+    * @param _tokenName String with the name for the token used by share holders in the organization
+    * @param _tokenSymbol String with the symbol for the token used by share holders in the organization
+    * @param _id String with the name for org, will assign `[id].aragonid.eth`
+    * @param _holders Array of token holder addresses
+    * @param _stakes Array of token stakes for holders (token has 18 decimals, multiply token amount `* 10^18`)
+    * @param _financePeriod Initial duration for accounting periods, it can be set to zero in order to use the default of 30 days.
+    * @param _useAgentAsVault Boolean to tell whether to use an Agent app as a more advanced form of Vault app
+    */
+    function newTokenAndBaseInstance(
+        string _tokenName,
+        string _tokenSymbol,
+        string _id,
+        address[] _holders,
+        uint256[] _stakes,
+        uint64 _financePeriod,
+        bool _useAgentAsVault
+    )
+        external
+    {
+        newToken(_tokenName, _tokenSymbol);
+        newBaseInstance(_id, _holders, _stakes, _financePeriod, _useAgentAsVault);
+    }
+
+    function installDandelionApps(
+        string _id,
+        address[] tokenRequestAcceptedDepositTokens,
+        address timeLockToken,
+        uint256[3] timeLockSettings
+
+    )
+        external
+    {
+        _ensureBaseAppsCache();
+        Kernel dao = _popDaoCache();
+        ACL acl = ACL(dao.acl());
+        bool agentAsVault = _popAgentAsVaultCache();
+        (,,Finance finance) = _popBaseAppsCache();
+        (Voting dandelionVoting,,,) = _popDandelionAppsCache();
```

Comment:
I think the dandelionVoting needs to be included in the `_setupBaseApps` and then we will retrieve it with something like: `(,,Finance finance, Voting dandelionVoting) = _popBaseAppsCache();`